[BUG]  [JAVA][MVN] 6.5.0 Generation of  Set<List<String>> failed on toUrlQueryString

#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

<!--
Please follow the issue template below for bug reports.
Also please indicate in the issue title which language/library is concerned. Eg:  [BUG][JAVA] Bug generating foo with bar 
-->

##### Description

<!-- describe what is the question, suggestion or issue and why this is a problem for you. -->
generated java client cannot pass compilation, see example error below
```shell
> Task :compileJava FAILED
/Users/stevezhang/workspace/iceberg-rest-catalog/client/build/generated/src/main/java/com/my/rest/model/ListNamespaces200Response.java:158: error: not a statement
      for (List&lt;String&gt; _item : getNamespaces()) {
               ^
/Users/stevezhang/workspace/iceberg-rest-catalog/client/build/generated/src/main/java/com/my/rest/model/ListNamespaces200Response.java:158: error: not a statement
      for (List&lt;String&gt; _item : getNamespaces()) {
                              ^
/Users/stevezhang/workspace/iceberg-rest-catalog/client/build/generated/src/main/java/com/my/rest/model/ListNamespaces200Response.java:158: error: ')' expected
      for (List&lt;String&gt; _item : getNamespaces()) {
                                   ^
/Users/stevezhang/workspace/iceberg-rest-catalog/client/build/generated/src/main/java/com/my/rest/model/ListNamespaces200Response.java:158: error: illegal start of expression
      for (List&lt;String&gt; _item : getNamespaces()) {
                                    ^
/Users/stevezhang/workspace/iceberg-rest-catalog/client/build/generated/src/main/java/com/my/rest/model/ListNamespaces200Response.java:158: error: ';' expected
      for (List&lt;String&gt; _item : getNamespaces()) {
                                                     ^
5 errors
```

##### Actual generated Output
```java
    // add `namespaces` to the URL query string
    if (getNamespaces() != null) {
      int i = 0;
      for (List&lt;String&gt; _item : getNamespaces()) {
        joiner.add(String.format("%snamespaces%s%s=%s", prefix, suffix,
            "".equals(suffix) ? "" : String.format("%s%d%s", containerPrefix, i, containerSuffix),
            URLEncoder.encode(String.valueOf(_item), StandardCharsets.UTF_8).replaceAll("\\+", "%20")));
      }
      i++;
    }
```


##### Expected Output
```java
    // add `namespaces` to the URL query string
    if (getNamespaces() != null) {
      int i = 0;
      for (List<String> _item : getNamespaces()) {
        joiner.add(String.format("%snamespaces%s%s=%s", prefix, suffix,
            "".equals(suffix) ? "" : String.format("%s%d%s", containerPrefix, i, containerSuffix),
            URLEncoder.encode(String.valueOf(_item), StandardCharsets.UTF_8).replaceAll("\\+", "%20")));
      }
      i++;
    }
```

##### openapi-generator version

6.5.0


##### OpenAPI declaration file content or url

  --> https://gist.github.com/dramaticlly/7d1f9ce2171f6e8a348b0aaee1c76c35

##### Generation Details

gradle task which use maven plugin to generate 

```gradle
task generateJavaClient(type: GenerateTask) {
	generatorName = "java"
	inputSpec = "$rootDir/rest-catalog.yaml"
	outputDir = "$buildDir/generated"
	apiPackage = "com.my.rest.api"
	invokerPackage = ""com.my.rest.invoker"
	modelPackage = "com.my.rest.model"
	configOptions = [
			disallowAdditionalPropertiesIfNotPresent: "false"
	]
}
```

##### Steps to reproduce

1. ./gradlew generateJavaClient
2. ./gradlew clean build

##### Related issues/PRs

<!-- has a similar issue/PR been reported/opened before? Please do a search in https://github.com/openapitools/openapi-generator/issues?utf8=%E2%9C%93&q=is%3Aissue%20 -->

##### Suggest a fix

<!-- if you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit), or simply make a suggestion -->

--- a/modules/openapi-generator/src/main/resources/Java/libraries/native/anyof_model.mustache
+++ b/modules/openapi-generator/src/main/resources/Java/libraries/native/anyof_model.mustache
@@ -237,7 +237,7 @@ public class {{classname}} extends AbstractOpenApiSchema{{#vendorExtensions.x-im
         {{#uniqueItems}}
         if (getActualInstance() != null) {
           int i = 0;
-          for ({{items.dataType}} _item : ({{{dataType}}})getActualInstance()) {
+          for ({{{items.dataType}}} _item : ({{{dataType}}})getActualInstance()) {
             joiner.add(String.format("%s{{baseName}}%s%s=%s", prefix, suffix,
                 "".equals(suffix) ? "" : String.format("%s%d%s", containerPrefix, i, containerSuffix),
                 URLEncoder.encode(String.valueOf(_item), StandardCharsets.UTF_8).replaceAll("\\+", "%20")));
@@ -260,7 +260,7 @@ public class {{classname}} extends AbstractOpenApiSchema{{#vendorExtensions.x-im
         {{#uniqueItems}}
         if (getActualInstance() != null) {
           int i = 0;
-          for ({{items.dataType}} _item : ({{{dataType}}})getActualInstance()) {
+          for ({{{items.dataType}}} _item : ({{{dataType}}})getActualInstance()) {
             if (_item != null) {
               joiner.add(_item.toUrlQueryString(String.format("%s{{baseName}}%s%s", prefix, suffix,
                   "".equals(suffix) ? "" : String.format("%s%d%s", containerPrefix, i, containerSuffix))));
@@ -284,7 +284,7 @@ public class {{classname}} extends AbstractOpenApiSchema{{#vendorExtensions.x-im
         {{#uniqueItems}}
         if (getActualInstance() != null) {
           int i = 0;
-          for ({{items.dataType}} _item : ({{{dataType}}})getActualInstance()) {
+          for ({{{items.dataType}}} _item : ({{{dataType}}})getActualInstance()) {
             if (_item != null) {
               joiner.add(String.format("%s{{baseName}}%s%s=%s", prefix, suffix,
                   "".equals(suffix) ? "" : String.format("%s%d%s", containerPrefix, i, containerSuffix),

--- a/modules/openapi-generator/src/main/resources/Java/libraries/native/oneof_model.mustache
+++ b/modules/openapi-generator/src/main/resources/Java/libraries/native/oneof_model.mustache
@@ -270,7 +270,7 @@ public class {{classname}} extends AbstractOpenApiSchema{{#vendorExtensions.x-im
         {{#uniqueItems}}
         if (getActualInstance() != null) {
           int i = 0;
-          for ({{items.dataType}} _item : ({{{dataType}}})getActualInstance()) {
+          for ({{{items.dataType}}} _item : ({{{dataType}}})getActualInstance()) {
             joiner.add(String.format("%s{{baseName}}%s%s=%s", prefix, suffix,
                 "".equals(suffix) ? "" : String.format("%s%d%s", containerPrefix, i, containerSuffix),
                 URLEncoder.encode(String.valueOf(_item), StandardCharsets.UTF_8).replaceAll("\\+", "%20")));
@@ -293,7 +293,7 @@ public class {{classname}} extends AbstractOpenApiSchema{{#vendorExtensions.x-im
         {{#uniqueItems}}
         if (getActualInstance() != null) {
           int i = 0;
-          for ({{items.dataType}} _item : ({{{dataType}}})getActualInstance()) {
+          for ({{{items.dataType}}} _item : ({{{dataType}}})getActualInstance()) {
             if (_item != null) {
               joiner.add(_item.toUrlQueryString(String.format("%s{{baseName}}%s%s", prefix, suffix,
                   "".equals(suffix) ? "" : String.format("%s%d%s", containerPrefix, i, containerSuffix))));
@@ -317,7 +317,7 @@ public class {{classname}} extends AbstractOpenApiSchema{{#vendorExtensions.x-im
         {{#uniqueItems}}
         if (getActualInstance() != null) {
           int i = 0;
-          for ({{items.dataType}} _item : ({{{dataType}}})getActualInstance()) {
+          for ({{{items.dataType}}} _item : ({{{dataType}}})getActualInstance()) {
             if (_item != null) {
               joiner.add(String.format("%s{{baseName}}%s%s=%s", prefix, suffix,
                   "".equals(suffix) ? "" : String.format("%s%d%s", containerPrefix, i, containerSuffix),

--- a/modules/openapi-generator/src/main/resources/Java/libraries/native/pojo.mustache
+++ b/modules/openapi-generator/src/main/resources/Java/libraries/native/pojo.mustache
@@ -397,7 +397,7 @@ public class {{classname}} {{#parent}}extends {{{.}}} {{/parent}}{{#vendorExtens
     {{#uniqueItems}}
     if ({{getter}}() != null) {
       int i = 0;
-      for ({{items.dataType}} _item : {{getter}}()) {
+      for ({{{items.dataType}}} _item : {{getter}}()) {
         joiner.add(String.format("%s{{baseName}}%s%s=%s", prefix, suffix,
             "".equals(suffix) ? "" : String.format("%s%d%s", containerPrefix, i, containerSuffix),
             URLEncoder.encode(String.valueOf(_item), StandardCharsets.UTF_8).replaceAll("\\+", "%20")));
@@ -420,7 +420,7 @@ public class {{classname}} {{#parent}}extends {{{.}}} {{/parent}}{{#vendorExtens
     {{#uniqueItems}}
     if ({{getter}}() != null) {
       int i = 0;
-      for ({{items.dataType}} _item : {{getter}}()) {
+      for ({{{items.dataType}}} _item : {{getter}}()) {
         if (_item != null) {
           joiner.add(_item.toUrlQueryString(String.format("%s{{baseName}}%s%s", prefix, suffix,
               "".equals(suffix) ? "" : String.format("%s%d%s", containerPrefix, i, containerSuffix))));
@@ -444,7 +444,7 @@ public class {{classname}} {{#parent}}extends {{{.}}} {{/parent}}{{#vendorExtens
     {{#uniqueItems}}
     if ({{getter}}() != null) {
       int i = 0;
-      for ({{items.dataType}} _item : {{getter}}()) {
+      for ({{{items.dataType}}} _item : {{getter}}()) {
         if (_item != null) {
           joiner.add(String.format("%s{{baseName}}%s%s=%s", prefix, suffix,
               "".equals(suffix) ? "" : String.format("%s%d%s", containerPrefix, i, containerSuffix),

--- a/modules/openapi-generator/src/main/resources/Java/pojo.mustache
+++ b/modules/openapi-generator/src/main/resources/Java/pojo.mustache
@@ -393,7 +393,7 @@ public class {{classname}} {{#parent}}extends {{{.}}} {{/parent}}{{#vendorExtens
     {{#uniqueItems}}
     if ({{getter}}() != null) {
       int i = 0;
-      for ({{items.dataType}} _item : {{getter}}()) {
+      for ({{{items.dataType}}} _item : {{getter}}()) {
         try {
           joiner.add(String.format("%s{{baseName}}%s%s=%s", prefix, suffix,
               "".equals(suffix) ? "" : String.format("%s%d%s", containerPrefix, i, containerSuffix),
@@ -426,7 +426,7 @@ public class {{classname}} {{#parent}}extends {{{.}}} {{/parent}}{{#vendorExtens
     {{#uniqueItems}}
     if ({{getter}}() != null) {
       int i = 0;
-      for ({{items.dataType}} _item : {{getter}}()) {
+      for ({{{items.dataType}}} _item : {{getter}}()) {
         if (_item != null) {
           joiner.add(_item.toUrlQueryString(String.format("%s{{baseName}}%s%s", prefix, suffix,
               "".equals(suffix) ? "" : String.format("%s%d%s", containerPrefix, i, containerSuffix))));
@@ -450,7 +450,7 @@ public class {{classname}} {{#parent}}extends {{{.}}} {{/parent}}{{#vendorExtens
     {{#uniqueItems}}
     if ({{getter}}() != null) {
       int i = 0;
-      for ({{items.dataType}} _item : {{getter}}()) {
+      for ({{{items.dataType}}} _item : {{getter}}()) {
         if (_item != null) {
           try {
             joiner.add(String.format("%s{{baseName}}%s%s=%s", prefix, suffix,

